GDJS/tests/tests/hot-reloader.js

```diff
@@ -36,6 +36,410 @@ describe('gdjs.HotReloader.deepEqual', () => {
+    //@ts-ignore
+    const scene = runtimeGame.getSceneStack().getCurrentScene();
+
+    const newProjectData = createProjectData({
+      layouts: [
+        createSceneData({
+          instances: [
+            { persistentUuid: '1', x: 555, y: 678 },
+            { persistentUuid: '2', x: 222, y: 234 },
+          ],
+        }),
+      ],
+    });
+
+    await hotReloader._hotReloadRuntimeGame(
+      oldProjectData,
+      newProjectData,
+      [],
+      runtimeGame
+    );
+
+    const instances = scene.getInstancesOf('MyObject');
+    expect(instances.length).to.be(2);
+    expect(instances[0].getX()).to.be(555);
+    expect(instances[0].getY()).to.be(678);
+    expect(instances[1].getX()).to.be(222);
+    expect(instances[1].getY()).to.be(234);
+  });
+
+  it('can change the image of a sprite object of a scene at hot-reload', async () => {
+    const oldProjectData = createProjectData({
+      layouts: [
+        createSceneData({
+          instances: [{ persistentUuid: '1' }, { persistentUuid: '2' }],
+          objects: [createSpriteData({ name: 'MyObject', image: 'ResourceA' })],
+        }),
+      ],
+    });
+    const runtimeGame = new gdjs.RuntimeGame(oldProjectData);
+    const hotReloader = new gdjs.HotReloader(runtimeGame);
+    runtimeGame.areSceneAssetsReady = (sceneName) => true;
+    runtimeGame._sceneStack.push('Scene1');
+    /** @type gdjs.RuntimeScene */
+    //@ts-ignore
+    const scene = runtimeGame.getSceneStack().getCurrentScene();
+
+    const newProjectData = createProjectData({
+      layouts: [
+        createSceneData({
+          instances: [{ persistentUuid: '1' }, { persistentUuid: '2' }],
+          objects: [createSpriteData({ name: 'MyObject', image: 'ResourceB' })],
+        }),
+      ],
+    });
+
+    await hotReloader._hotReloadRuntimeGame(
+      oldProjectData,
+      newProjectData,
+      [],
+      runtimeGame
+    );
+
+    const instances = scene.getInstancesOf('MyObject');
+    expect(instances.length).to.be(2);
+    expect(getSpriteImage(instances[0])).to.be('ResourceB');
+    expect(getSpriteImage(instances[1])).to.be('ResourceB');
+  });
+
+  it('can move instances inside a custom object at hot-reload', async () => {
+    const oldProjectData = createProjectData({
+      layouts: [
+        createSceneData({
+          instances: [{ persistentUuid: '1', name: 'MyCustomObject' }],
+          objects: [defaultCustomObject],
+        }),
+      ],
+      eventsBasedObjects: [
+        createEventsBasedObjectData({
+          name: 'MyCustomObject',
+          instances: [
+            { persistentUuid: '11', x: 111, y: 123 },
+            { persistentUuid: '12', x: 222, y: 234 },
+          ],
+        }),
+      ],
+    });
+    const runtimeGame = new gdjs.RuntimeGame(oldProjectData);
+    const hotReloader = new gdjs.HotReloader(runtimeGame);
+    runtimeGame.areSceneAssetsReady = (sceneName) => true;
+    runtimeGame._sceneStack.push('Scene1');
+    /** @type gdjs.RuntimeScene */
```

Comment:
```suggestion
    /** @type {gdjs.RuntimeScene} */
```

Reply:
But this will be useless once you throw when the null case is found, thanks to TS type refining.